### DISCUSSION TITLE
Migrating postgres data from Heroku to Dokku

### DISCUSSION THREAD

**Discussion by venkibseng** [dokku/dokku#5327] [General]
Hi,

As Heroku finally dropped the bombshell of discontinuing the Free Dynos, I am looking to move my app from Heroku to Dokku. 

It is a very simple Django app, it uses Postgress and Heroku Scheduler. 

I came across https://dokku.com/docs/deployment/application-deployment/ which is a great starting point. But it is silent on database migrations (especially data migration).

Any pointers/tools to migrate data from Postgress in Heroku to Dokku. 

I suppose I can dump the data to .sql and import it into Dokku's Postgress.  But is there any better solution ?

**Comment by josegonzalez** [comment]
There aren't any built-in ways to migrate data because thats a complex topic and I don't want to pretend to know the vagaries of every datastore 😅 . Dump + restore is probably the easiest way to do it, but the postgres community may have other ways of doing the migration that are preferred.

**Comment by abulka** [comment]
> As Heroku finally dropped the bombshell of discontinuing the Free Dynos (for whatever rubbish excuse they could give)

The developer community is not happy with this news - Twitter is all abuzz, looking for alternatives - I've dropped a few tweets mentioning dokku. I wonder what the impact of this news will be on the dokku project!

**Comment by josegonzalez** [comment]
One note:

> As Heroku finally dropped the bombshell of discontinuing the Free Dynos (for whatever rubbish excuse they could give), I am looking to move my app from Heroku to Dokku.

I don't think this is an appropriate statement to make. It's a bit childish and also I think goes against the spirit of our Code of Conduct.

Heroku has every right to change how they do business. There is certainly some calculus where they make money from free users, but with the rise of cryptocurrency and general platform abuse, I can imagine that the tradeoff just isn't worth it. If they decide that your free usage isn't bringing them any revenue - and it isn't! - then it is definitely within their rights to turn off their free tier.

Lets avoid any commentary in this vein gong forward :)

**Reply by venkibseng** [reply]
Thanks for the feedback. I have now updated the OP.
